docs/madr/decisions/012-health-check-policy.md

```diff
@@ -0,0 +1,180 @@
+# Health Check Policy
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4735
+
+## Context and Problem Statement
+
+We want to create a new policy for Health Checking that uses TargetRef [matching](https://github.com/kumahq/kuma/blob/22c157d4adac7f518b1b49939c7e9ea4d2a1876c/docs/madr/decisions/005-policy-matching.md).
+
+We want to extend the current list of supported protocols (TCP, HTTP) with GRPC.
+
+As a reminder: this is active health checking so this policy will attach to an outbound.
+
+## Specification
+
+### Matching
+
+#### From level
+
+The way active health checking works is by sending outgoing requests, I don't think it makes sense to have "from" section.
+
+#### Top & to level
+
+`Top` and `to` levels look to me as mirror cases, so I think it's worth talking about them together.
+
+`MeshGatewayRoute` and `MeshHTTPRoute` do not make sense as on its own they do not have "health",
+also Envoy does set health on a route only per Cluster/Endpoint.
+
+`MeshService` is a natural fit, but I think we should be careful when allowing other targets combinations.
+
+##### Option 1 - allow everything
+
+###### Pros
+
+- Flexibility
```

Comment:
What gives us this flexibility? Do you have any example use cases that this flexibility will solve and third option with only `MeshService` won't solve

Reply:
It's just a flexibility of being able to select different dataplanes. If you don't have `Mesh` / `MeshSubset` and you have **10 services** you'd need to define **10 entries** in `to` or **10 policies** with top level target ref (instead of one with `Mesh`). It's not that it's not possible it's just a lot of repetitive config.